### DISCUSSION TITLE
is it possible to disable `history-words` either globally, or for certain commands?

### DISCUSSION THREAD

**Discussion by vitaly** [marlonrichert/zsh-autocomplete#305] [Q&A] [answered]

**Comment by marlonrichert** [comment, marked as answer]
Yes, you can control this through the [`tag-order` style](https://zsh.sourceforge.io/Doc/Release/Completion-System.html#index-tag_002dorder_002c-completion-style): 
```zsh 
zstyle ':completion:*' tag-order '! history-words' -
```

**Reply by vitaly** [reply]
This certainly works and disables it globally. but if I try to do it for `':completion:*:cd:*'`, for some reason it doesn't. is there a way to target `cd` specifically?

**Reply by marlonrichert** [reply]
No, that’s not possible. `cd` gets set as a context inside [`_complete`](https://github.com/zsh-users/zsh/blob/master/Completion/Base/Completer/_complete) only. [`_history`](https://github.com/zsh-users/zsh/blob/master/Completion/Base/Completer/_history) doesn’t go into that kind of detail.

**Reply by marlonrichert** [reply]
FYI: With d7a0899cc692687b902abd164df610dded9b9233, I've now disabled history word completion completely, as it can cause serious performance hits.
